easybuild/easyconfigs/j/jax/jax-0.2.18-foss-2020b.eb

```diff
@@ -0,0 +1,87 @@
+# This file is an EasyBuild reciPY as per https://github.com/easybuilders/easybuild
+# Author: Denis Kristak
+easyblock = 'PythonBundle'
+
+name = 'jax'
+version = '0.2.18'
+local_jaxlib_ver = '0.1.70'
+homepage = 'https://pypi.python.org/pypi/jax'
+description = """Composable transformations of Python+NumPy programs: differentiate, vectorize, JIT to GPU/TPU, and
+more"""
+
+toolchain = {'name': 'foss', 'version': '2020b'}
+
+# downloading TF tarball for better control of what Bazel will use during installation of Jax
+local_tf_tarball_version = '2.5.0'
+
+builddependencies = [
+    ('Bazel', '3.7.2'),
+]
+
+dependencies = [
+    ('Python', '3.8.6'),
+    ('SciPy-bundle', '2020.11'),
+    ('Cython', '0.29.22'),
+    ('opt-einsum', '3.3.0'),
+    ('flatbuffers-python', '1.12'),
+]
+
+local_tf_builddir = "%%(builddir)s/tensorflow-%s" % local_tf_tarball_version
+
+local_jax_preinstallopts = "sed -i -e 's$pathToSed$%s$g' WORKSPACE && " % local_tf_builddir
+
+local_jax_build_cmd = "sed -i -e 's$pathToSed$%s$g' WORKSPACE && " % local_tf_builddir
+local_jax_build_cmd += 'python build/build.py '
+local_jax_build_cmd += '--bazel_startup_options="--output_user_root=%(builddir)s" '
+local_jax_build_cmd += '--bazel_path="$EBROOTBAZEL/bin/bazel" '
+local_jax_build_cmd += '--bazel_options=--subcommands '
+local_jax_build_cmd += '--bazel_options=--jobs=1 --bazel_options=--action_env=PYTHONPATH '
+local_jax_build_cmd += '--bazel_options=--action_env=EBPYTHONPREFIXES'
+
+
+exts_list = [
+    ('opt-einsum', '3.3.0', {
+        'source_tmpl': 'opt_einsum-%(version)s.tar.gz',
+        'modulename': 'opt_einsum',
+    }),
+    ('absl-py', '0.13.0', {
+        'modulename': 'absl'
+    })]
+
+default_component_specs = {
+    'sources': [SOURCE_TAR_GZ],
+    'start_dir': '%(name)s-%(version)s',
+}
+
+components = [
+    ('jaxlib', local_jaxlib_ver, {
+        'easyblock': 'PythonPackage',
+        'sources': [
+            'jaxlib-v%s.zip' % local_jaxlib_ver,
+            {
+                'download_filename': 'v%s.tar.gz' % local_tf_tarball_version,
+                'filename': 'tensorflow_v%s.tar.gz' % local_tf_tarball_version,
+            }
+        ],
+        'source_urls': [
+            'https://github.com/google/jax/archive/',
+            'https://github.com/tensorflow/tensorflow/archive/'
+        ],
+        'patches': [
+            ('jaxlib-%s-correct_libraries.patch' % local_jaxlib_ver, 1),
+            ('jaxlib-%s-controlled_tf_tarball.patch' % local_jaxlib_ver, '../tensorflow-%s' % local_tf_tarball_version),
+        ],
+        'start_dir': 'jax-jaxlib-v%s' % local_jaxlib_ver,
+        'preinstallopts': local_jax_preinstallopts,
+        'use_pip': True,
+        'sanity_pip_check': True,
+        'download_dep_fail': True,
+        'buildcmd': local_jax_build_cmd,
+        'install_src': 'dist/*.whl',
+    }),
+    ('jax', '0.2.18', {
```

Comment:
I'd suggest moving that to exts_list, then you can keep opt-einsum there

Reply:
done